hopsworks-common/src/main/java/io/hops/hopsworks/common/serving/util/KafkaServingHelper.java

```diff
@@ -160,28 +165,29 @@ private ProjectTopics setupKafkaTopic(Project project, ServingWrapper servingWra
       Settings.INFERENCE_SCHEMAVERSION);
 
     ProjectTopics pt = null;
-    pt = kafkaFacade.createTopicInProject(project, topicDTO);
+    SchemaTopics schema =
+      kafkaFacade.findSchemaByNameAndVersion(topicDTO.getSchemaName(), topicDTO.getSchemaVersion()).orElseThrow(() ->
+        new KafkaException(RESTCodes.KafkaErrorCode.SCHEMA_NOT_FOUND, Level.FINE, "topic: " + topicDTO.getName()));
+  
+    pt = kafkaFacade.createTopicInProject(project, topicDTO, schema);
```

Comment:
@kai-chi we can make life easier for the client by having `kafkaFacade.createTopicInProject(project, topicDTO);` which does 
```
    SchemaTopics schema =
      kafkaFacade.findSchemaByNameAndVersion(topicDTO.getSchemaName(), topicDTO.getSchemaVersion()).orElseThrow(() ->
        new KafkaException(RESTCodes.KafkaErrorCode.SCHEMA_NOT_FOUND, Level.FINE, "topic: " + topicDTO.getName()));
```